electrumx/server/db.py

```diff
@@ -409,6 +410,10 @@ def lookup_hashXs():
             for each prevout.
             '''
             def lookup_hashX(tx_hash, tx_idx):
+                if tx_hash == TxInput.ZERO and tx_idx == TxInput.MINUS_1:
+                    # No hashX for coinbase-like inputs
+                    return None, None
+
```

Comment:
The changes look fine to me.  Can you explain this though?  This will be impacting utxo lookup for every other coin.  Is this just an optimization for Decred, or is it a necessary part of the semantics?  Is there some other way of achieving it?

Reply:
This code in db.py is not where I thought it was; I thought it was in a hot loop but it's not.

Can you add a comment about why this is needed?  It's not obvious on a casual read that it's Decred-specific.  Also I think I will simpify the is_coinbase code in tx.py to omit the tx hash comparison, the tx_idx comparison is enough.  Can you remove the tx_hash comparison from the above snippet?

I'll merge this with those 2 minor changes.  Thanks a lot.

Reply:
Removed completely this code.